### PR TITLE
[Profiler] Provides an alternative stackwalk

### PR DESCRIPTION
**Pull request by gleocadie**
## Summary of changes

Provide an alternative stackwalk mechanism.

## Reason for change

We would like to compare the old/manual way  and the new way to stackwalk a thread on linux. The goal is to continuously see improvements or degradation, and also propose an alternative to investigate customer issue.

## Implementation details

- Add a new internal env var to select the stackwalk algorithm
- Switch between the two algorithm depending on the env var.
- Add benchmark tests.

### CODE REVIEW

File: profiler/test/Datadog.Profiler.Native.Tests/LinuxStackFramesCollectorTest.cpp [hunk 1]
```diff
@@ -290,7 +293,11 @@ class LinuxStackFramesCollectorFixture : public ::testing::Test
 TEST_F(LinuxStackFramesCollectorFixture, CheckSamplingThreadCollectCallStack)
 {
     auto* signalManager = GetSignalManager();
-    auto collector = LinuxStackFramesCollector(signalManager);
+
+    auto [configuration, mockConfiguration] = CreateConfiguration();
+    EXPECT_CALL(mockConfiguration, UseBacktrace2()).WillOnce(Return(true));
```
Comment (chrisnas) [inline profiler/test/Datadog.Profiler.Native.Tests/LinuxStackFramesCollectorTest.cpp:298]: Would it be possible (like in C#) to have the same test to run once with backtrace2 and once with the previous implementation?
That way, both implementations would be tested

**Review by chrisnas — APPROVED** [review]
See if it is possible to run Linux stack walk tests with the 2 implementations

File: profiler/test/Datadog.Profiler.Native.Tests/LinuxStackFramesCollectorTest.cpp (hunk 1 above)
Comment (gleocadie) [reply profiler/test/Datadog.Profiler.Native.Tests/LinuxStackFramesCollectorTest.cpp:298]: oh good one.
I was adding test for the configuration but forgot this one.

File: profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp [hunk 2]
```diff
@@ -794,6 +792,23 @@ void CorProfilerCallback::ConfigureDebugLog()
...
+        auto envVarValue = shared::GetEnvironmentValue(name); \
+        if (!envVarValue.empty())                             \
+        {                                                     \
+            Log::Info("  ", name, ": ", envVarValue);         \
+        }                                                     \
+    }
+
+void CorProfilerCallback::PrintEnvironmentVariables()
+{
+    // TODO: add more env vars values
```
Comment (chrisnas) [inline profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp:806]: This is a good idea: we should also provide the command line like the Tracer is doing

**Review by chrisnas — APPROVED** [review]
LGTM

**Pull request merged**